provision-osx/tasks/install_xcode.yml

```diff
@@ -9,15 +9,6 @@
   register: user_groups
 
 -
-  name: Ensure passwordless sudo for user
```

Comment:
Oh dear! I thought I had removed this ages ago! 😨  

There are a few other blocks that can be removed here `Get current user groups`,  `Remove passwordless sudo for user`, `Restore user groups`.

Apologies for not catching this earlier, these should have never been included in the commit.